### PR TITLE
Fix mz ml autodetect

### LINKED ISSUE #2563: IDFileConverter detects the input type according to the file extension
Therefore, the tool is not easy to integrate into frameworks like Galaxy. Integration would be improved by an additional param where the input file format can be determined. Or is it also possible to read the file format from metadata somehow?

### PR DESCRIPTION
**Pull request by cbielow**
fixes #2563

FileHandler -- use capital 'M' in MzIdentML as required by the schema

### CODE REVIEW

**Comment by jpfeuffer** [comment]
Great. +1 if passes

**Comment by timosachsenberg** [comment]
+1 one timeout

File: src/openms/source/FORMAT/FileHandler.cpp [hunk 1]
```diff
@@ -238,7 +238,7 @@ namespace OpenMS
       return FileTypes::MZML;
 
     //"analysisXML" aka. mzid (all lines)
-    if (all_simple.hasSubstring("<mzIdentML"))
+    if (all_simple.hasSubstring("<MzIdentML"))
```
Comment (hroest) [inline src/openms/source/FORMAT/FileHandler.cpp]: why not just use `all_simple.toLower()` and then compare?

Comment (cbielow) [inline src/openms/source/FORMAT/FileHandler.cpp]: I had that in mind as well, but technically, XML is case sensitive and any other term is not valid XML -- technically.

**Pull request merged**